src/bsp/mount_fatfs.c

```diff
@@ -35,43 +44,37 @@ int fatfs_mount()
 {
     printk("Mounting fatfs...");
 
-    uint32_t block_count;
-    uint32_t block_size;
-
     int ret = disk_access_init(DISK_DRIVE_NAME);
     if (ret) {
-        printk("[fatfs] access init failed (%d)\n", ret);
+        printk("[fatfs] Access init failed (%d)\n", ret);
 
         return ret;
     }
 
-    ret = disk_access_ioctl(DISK_DRIVE_NAME, DISK_IOCTL_GET_SECTOR_COUNT, &block_count);
+    ret = fs_mount(&mp_aos);
     if (ret) {
-        printk("[fatfs] get sector count failed (%d)\n", ret);
+        printk("[fatfs] Aos disk mount failed (%d)\n", ret);
 
         return ret;
     }
 
-    ret = disk_access_ioctl(DISK_DRIVE_NAME, DISK_IOCTL_GET_SECTOR_SIZE, &block_size);
+    printk("[fatfs] Aos disk mounted at %s\n", CONFIG_AOS_DISK_MOUNT_POINT);
+
+    ret = fs_mount(&mp_tmp);
     if (ret) {
-        printk("[fatfs] get sector size failed (%d)\n", ret);
+        printk("[fatfs] TMP disk mount failed (%d)\n", ret);
 
         return ret;
     }
 
-    mp.mnt_point = CONFIG_AOS_DISK_MOUNT_POINT;
+    printk("[fatfs] TMP disk mounted at %s\n", DISK_TMP_MOUNT_PT);
 
-    ret = fs_mount(&mp);
+    ret = fs_mkdir(DISK_TMP_MOUNT_PT "/tmp");
```

Comment:
shouldn't mkdir be called before mount(tmp)?

Reply:
no. We should mount fs and then create /tmp dir there.